Doesn't seem possible to use --warn-unused-ignores without --strict

I assume the parser is looking for errors, not warnings, in this case.  When trying to incrementally improve things on an existing project, `--strict` is a pretty big jump.

Example, each time you run it appears to append another "FIX ME" comment.

```sh-session
$ mkdir /tmp/foo
$ cd /tmp/foo
$ python -m venv .venv
$ . .venv/bin/activate
$ pip install mypy mypy-upgrade
$ printf 'x: int = "a"\ny: str = x\n" > foo.py
$ mypy foo.py > t; mypy-upgrade < t
$ cat foo.py 
x: int = "a"  # type: ignore[assignment] # FIX ME
y: str = x  # type: ignore[assignment] # FIX ME
$ sed -i.bak -e 's/int/str/' foo.py
$ cat foo.py 
x: str = "a"  # type: ignore[assignment] # FIX ME
y: str = x  # type: ignore[assignment] # FIX ME
$ mypy foo.py
Success: no issues found in 1 source file
$ mypy foo.py --warn-unused-ignores
foo.py:1: error: Unused "type: ignore" comment  [unused-ignore]
foo.py:2: error: Unused "type: ignore" comment  [unused-ignore]
Found 2 errors in 1 file (checked 1 source file)
$ mypy foo.py > t; mypy-upgrade < t
$ cat foo.py 
x: str = "a"  # type: ignore  # FIX ME # FIX ME
y: str = x  # type: ignore  # FIX ME # FIX ME
$ mypy foo.py --warn-unused-ignores > t; mypy-upgrade < t
$ cat foo.py
x: str = "a"  # type: ignore  # FIX ME # FIX ME # FIX ME
y: str = x  # type: ignore  # FIX ME # FIX ME # FIX ME
```

--- a/src/mypy_upgrade/editing.py
+++ b/src/mypy_upgrade/editing.py
@@ -73,14 +73,15 @@ def remove_unused_type_ignore_comments(
         r"#\s*type\s*:\s*ignore(\[(?P<error_code>[a-z, \-]+)\])?"
     )
     match = type_ignore.search(comment)
+    if "*" in codes_to_remove:
+        return type_ignore.sub("", comment)
+
     old_codes = match.group("error_code") if match is not None else ""
 
     if not old_codes:
         return comment
 
-    if "*" in codes_to_remove or all(
-        code.strip() in codes_to_remove for code in old_codes.split(",")
-    ):
+    if all(code.strip() in codes_to_remove for code in old_codes.split(",")):
         return type_ignore.sub("", comment)
 
     new_codes = old_codes

--- a/src/mypy_upgrade/silence.py
+++ b/src/mypy_upgrade/silence.py
@@ -151,17 +151,20 @@ def create_suppression_comment(
         comment=comment, codes_to_remove=to_remove
     )
     formatted_comment = format_type_ignore_comment(comment=pruned_comment)
-    suppression_comment = add_type_ignore_comment(
-        comment=formatted_comment,
-        error_codes=to_add,
-    )
-    if fix_me:
-        suppression_comment += f" # {fix_me}"
+    if to_add:
+        suppression_comment = add_type_ignore_comment(
+            comment=formatted_comment,
+            error_codes=to_add,
+        )
+        if fix_me:
+            suppression_comment += f" # {fix_me}"
 
-    if description_style == "full" and descriptions:
-        suppression_comment += f" # {', '.join(descriptions)}"
+        if description_style == "full" and descriptions:
+            suppression_comment += f" # {', '.join(descriptions)}"
 
-    return suppression_comment
+        return suppression_comment
+    else:
+        return formatted_comment
 
 
 def _writelines(*, file: TextIO, lines: Iterable[CommentSplitLine]) -> int:

